brave/src/main/java/brave/Tracer.java

```diff
@@ -223,10 +231,14 @@ public Span newChild(TraceContext parent) {
     if (Boolean.FALSE.equals(parent.sampled())) {
       return NoopSpan.create(parent);
     }
-    return toSpan(nextContext(parent, parent));
+    return toSpan(nextContext(null, parent, parent));
   }
 
-  TraceContext nextContext(@Nullable TraceContext parent, SamplingFlags samplingFlags) {
+  TraceContext nextContext(
```

Comment:
Since the actual precedence is `TraceContext` and then `TraceIdContext`, what about reflecting this in the signature?

Reply:
good idea. done